File: src/test/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderImageRepositoryTest.java
package uk.gov.justice.hmpps.prison.repository.jpa.repository;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.test.context.ActiveProfiles;
import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderImage;

import static org.assertj.core.api.Assertions.assertThat;
import static org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace.NONE;

@DataJpaTest
@ActiveProfiles("test")

@AutoConfigureTestDatabase(replace = NONE)
public class OffenderImageRepositoryTest {

    @Autowired
    private OffenderImageRepository repository;

    @Test
    public void getImagesByOffenderNumber() {

        var images = repository.getImagesByOffenderNumber("A1234AA");

        assertThat(images).hasSize(1);
        assertThat(images).extracting(OffenderImage::getOffenderImageId).containsOnly(-1L);
    }

    @Test
    public void getImagesByOffenderNumberReturnsEmptyForOffenderNumberNotFound() {

        assertThat(repository.getImagesByOffenderNumber("unknown")).isEmpty();
    }
}

File: src/test/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderImageRepositoryTest.kt
package uk.gov.justice.hmpps.prison.repository.jpa.repository

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.data.jpa.test.autoconfigure.DataJpaTest
import org.springframework.boot.jdbc.test.autoconfigure.AutoConfigureTestDatabase
import org.springframework.test.context.ActiveProfiles

@DataJpaTest
@ActiveProfiles("test")
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
class OffenderImageRepositoryTest(
  @Autowired private val repository: OffenderImageRepository,
) {
  @Test
  fun getImagesByOffenderNumber() {
    val images = repository.getImagesByOffenderNumber("A1069AA")

    assertThat(images).hasSize(4)
    assertThat(images)
      .extracting<Long, RuntimeException> { it.getId() }
      .containsOnly(-100L, -101L, -102L, -103L)
  }

  @Test
  fun getImagesByOffenderNumberReturnsEmptyForOffenderNumberNotFound() {
    assertThat(repository.getImagesByOffenderNumber("unknown")).isEmpty()
  }
}
